### DISCUSSION TITLE
Why is a Date needed in some simple indicators like Sma, Rsi etc?

### DISCUSSION THREAD

**Discussion by 1nfected** [DaveSkender/Stock.Indicators#1074] [Q&A]
I've just started exploring this library and started looking the Sma indicator.

I noticed it required a Date along with the actual value, but I didn't quite understand why.

Maybe there is a need (which I haven't figured out yet), but then in that case, do you think it would make sense to add an overload which doesn't required a Date and simply accepts an array/list of values and assumes it is in the correct order?

**Comment by DaveSkender** [comment]
As part of our [guiding principles](https://github.com/DaveSkender/Stock.Indicators/discussions/648), we wanted this library to be _*more usable*_ than legacy libraries like TA-LIB.  Array based APIs leave a tremendous amount of work for developers to transform quotes into and out of arrays and back into an end-use contextual point in time.

With that said, having an overload that takes plain arrays could be an option to explore; *want to submit a PR to demonstrate the concept*?
Side note: we do have an undocumented simpler tuple (date, value) input method for most indicators.

## Array-based libraries

```mermaid
flowchart TD
    OHLCV[standard OHLCV quotes] --> C1[transform quotes to arrays] --> LIB[an old library, like TA-LIB] --> C2[tranform arrays to time contexts] --> USAGE[Time-based end-use contexts]
```

## Our library

```mermaid
flowchart TD
    OHLCV[standard OHLCV quotes] --> LIB[our NuGet library API] --> USAGE[Time-based end-use contexts]
```
